sklearn/feature_extraction/tests/test_text.py

```diff
@@ -1611,7 +1612,12 @@ def test_tfidf_transformer_copy(csr_container):
     assert X_transform is not X_csr
 
     X_transform = transformer.transform(X_csr, copy=False)
-    assert X_transform is X_csr
+    # allow for config["sparse_interface"] to change output type
+    # there should be no data copied, but the `id` will change.
+    if _as_sparse(X_csr) is X_csr:
+        assert X_transform is X_csr
+    else:
+        assert X_transform is not X_csr
```

Comment:
Although the object identity is not the same, is the underlying data/indices/indexes pointing to the same data? If so, we can check the data?

Reply:
Yes -- the underlying data is the same. Good idea.  I've added a check for `indptr`.